src/hotspot/share/classfile/javaClasses.cpp

```diff
@@ -502,7 +502,8 @@ jchar* java_lang_String::as_unicode_string_or_null(oop java_string, int& length)
   return result;
 }
 
-unsigned int java_lang_String::hash_code(oop java_string) {
+template<bool update>
+unsigned int java_lang_String::hash_code_impl(oop java_string) {
```

Comment:
This looks like a really weird way to pass a parameter! Why do we want a template here?

Reply:
Hi David, thanks for the review. I replaced the template function with an inline function.